### PR TITLE
Compiler symbol fixes

### PR DESCRIPTION
**Pull request by sonneveld**
* Allow record member qualifiers to be in any order
* friendly names for types in error messages
* fix readonly crashes

### CODE REVIEW

File: Solutions/Compiler.Lib/Compiler.Lib.Test.vcproj [hunk 1]
```diff
@@ -192,6 +196,10 @@
 				RelativePath="..\..\Common\util\stream.cpp"
 				>
 			</File>
+			<File
+				RelativePath="..\..\Common\util\stream.cpp"
```
Comment (ivan-mogilko) [inline Solutions/Compiler.Lib/Compiler.Lib.Test.vcproj]: Why is this file added for the second time?

Comment (sonneveld) [inline Solutions/Compiler.Lib/Compiler.Lib.Test.vcproj]: good question. Will fix.
